WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -677,6 +693,61 @@ private extension EditableOrderViewModel {
         analytics.track(event: WooAnalyticsEvent.Orders.orderProductAdd(flow: flow.analyticsFlow))
     }
 
+    /// Adds a selected product variation (from the product variation list) to the order.
+    ///
+    func addProductVariationToOrder(_ variation: ProductVariation) {
+        let input = OrderSyncProductInput(product: .variation(variation), quantity: 1)
+        orderSynchronizer.setProduct.send(input)
+
+        analytics.track(event: WooAnalyticsEvent.Orders.orderProductAdd(flow: flow.analyticsFlow))
+    }
+
+    /// Removes a selected product (from the product list) to the order.
+    ///
+    func removeProductFromOrder(_ product: Product) {
+        guard let orderItem = orderSynchronizer.order.items.first(where: { $0.productID == product.productID }) else {
+            DDLogError("Unable to find product ID: \(product.productID) in Order")
+            return
+        }
+        removeItemFromOrder(orderItem)
+        analytics.track(event: WooAnalyticsEvent.Orders.orderProductRemove(flow: flow.analyticsFlow))
+    }
+
+    /// Removes a selected product variation (from the product variation list) to the order.
+    ///
+    func removeProductVariationFromOrder(_ variation: ProductVariation) {
+        guard let orderItem = orderSynchronizer.order.items.first(where: { $0.variationID == variation.productVariationID }) else {
+            DDLogError("Unable to find product variation ID: \(variation.productVariationID) (parent ID: \(variation.productID)) in Order")
+            return
+        }
+        removeItemFromOrder(orderItem)
+        analytics.track(event: WooAnalyticsEvent.Orders.orderProductRemove(flow: flow.analyticsFlow))
+    }
+
+    /// Handles a product by adding it to the Order if selected, or removing it from the Order if unselected
+    ///
+    func handleProduct(_ product: Product) {
```

Comment:
I would rename this function to specify what's doing with the product, `handleProduct` sounds too generic for me and I wouldn't know what's happening inside.

Reply:
Renamed to `addOrRemoveProductFromOrder` here: https://github.com/woocommerce/woocommerce-ios/pull/8943/commits/cb86a9852a0da3321f719b4a40662a1577e60a8d